hooks/useUpdateBodyClass.ts

```diff
@@ -0,0 +1,13 @@
+export const useUpdateBodyClass = () => {
```

Comment:
This shouldn't be a hook; it can be a simple util mainly because it uses 0 Reactive APIs.

Now, the name of this function is a little bit misleading; even tho it accepts a "string" it pretty much has specific logic for the "dark/light" theme.

**Also, remember this will not work correctly with Next.js when built. Next.js will override the body class list on re-renders.** (This is usually not noticeable with dev mode, but we shouldn't modify the body/document directly as this is an anti-pattern for React applications).

**If you want to properly manage the theme based on the state of the theme, you should:**

- Use the `useTheme` within the `theme.tsx` file
- Add the className for `.dark` or `.light` in [this](https://github.com/nodejs/nodejs.org/blob/main/theme.tsx#L53) line

That would be a clean solution that doesn't involve dom manipulation.

Reply:
>This shouldn't be a hook; it can be a simple util mainly because it uses 0 Reactive APIs.

 It was a normal function previously, I moved it into a hook after your suggestion


>If you want to properly manage the theme based on the state of the theme, you should:
>
>Use the useTheme within the theme.tsx file
>Add the className for .dark or .light in [this](https://github.com/nodejs/nodejs.org/blob/main/theme.tsx#L53) line
>That would be a clean solution that doesn't involve dom manipulation.

Sounds great 👍🏻, but I tried to change the theme using `data-theme` attribute which has been updated by the `next-themes` and it works great.

Reply:
Oh yes, I suggested to make it a hook because I thought you would move the whole useTheme to there 🙃

But it was my bad to not actually look carefully for what the function was doing 😅

For Next.js we want to avoid doing manual DOM mutations as Next rewrites stuff.